provider/core/src/resource.rs

```diff
@@ -14,6 +14,21 @@ use core::fmt::Write;
 use icu_locid::LanguageIdentifier;
 use writeable::{LengthHint, Writeable};
 
+/// A compact hash of a [`ResourceKey`]. Useful for keys in maps.
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone, Hash)]
+#[repr(transparent)]
+pub struct ResourceKeyHash([u8; 4]);
+
+impl ResourceKeyHash {
+    const fn compute_from_str(path: &str) -> Self {
+        Self(helpers::fxhash_32(path.as_bytes()).to_le_bytes())
+    }
+
+    pub const fn get(&self) -> [u8; 4] {
+        self.0
+    }
```

Comment:
Suggestion: Make this

```rust
pub const fn to_bytes(self) -> [u8; 4]
```

This is consistent with the design of `u32::to_le_bytes()` for example.